python/ray/ml/utils/remote_storage.py

```diff
@@ -3,14 +3,27 @@
 
 try:
     import fsspec
+
 except ImportError:
     fsspec = None
 
 try:
     import pyarrow
     import pyarrow.fs
+
+    class _CustomFSSpecHandler(pyarrow.fs.FSSpecHandler):
+        """Custom FSSpecHandler that avoids bugs in some fsspec implementations."""
+
+        def create_dir(self, path, recursive):
+            try:
+                # No `create_parents` argument
+                self.fs.mkdir(path)
```

Comment:
So it doesn't support create folders recursively now? What if the parent folder does not exist?

Reply:
No, the gs fsspec implementation does not accept a `create_parents` argument. As seen in the linked PR for gcsfs, it does not implement the fsspec base class correctly.

Reply:
GS does not have the notion of parent directories, everything is just an absolute path

Reply:
I see. Curious what is the difference between gs and gcs?

Reply:
There's no difference, these are the two registered URI protocols in fsspec: https://github.com/fsspec/filesystem_spec/blob/master/fsspec/registry.py#L107-L114